src/custom/state/appData/atoms.ts

```diff
@@ -0,0 +1,96 @@
+import { atom } from 'jotai'
+import { atomWithStorage } from 'jotai/utils'
+
+import {
+  AddAppDataToUploadQueueParams,
+  AppDataPendingToUpload,
+  AppDataInfo,
+  FlattenedAppDataFromUploadQueue,
+  RemoveAppDataFromUploadQueueParams,
+  UpdateAppDataOnUploadQueueParams,
+} from 'state/appData/types'
+import { buildAppDataRecordKey, parseAppDataRecordKey } from 'state/appData/utils'
+
+const INITIAL_APP_DATA_UPLOAD_QUEUE: AppDataPendingToUpload = {}
+
+/**
+ * Base atom that store the current appDataInfo
+ */
+export const appDataInfoAtom = atom<AppDataInfo | null>(null)
+
+/**
+ * Base atom that stores all appData pending to be uploaded
+ */
+export const appDataUploadQueueAtom = atomWithStorage(
+  'appDataUploadQueue', // local storage key
+  INITIAL_APP_DATA_UPLOAD_QUEUE
```

Comment:
not sure it adds a lot of value to creat this const, only used here, is just to make sure the right type is picked? because I believe u can specify the type too

Reply:
Initially I had the state nested split by chainId, where it made more sense.

Will update it